File: ImageFiltersJava/app/src/main/java/org/geeksforgeeks/demo/Adapter.java
package org.geeksforgeeks.demo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {

    private final List<Filters> items;
    private final OnFilterClickListener listener;

    public Adapter(List<Filters> items, OnFilterClickListener listener) {
        this.items = items;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_rv, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Filters filterItem = items.get(position);
        holder.effectTitle.setText(filterItem.getTitle());

        holder.effectRootView.setOnClickListener(v -> listener.onFilterClicked(filterItem));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView effectTitle;
        LinearLayout effectRootView;

        public ViewHolder(@NonNull View view) {
            super(view);
            effectTitle = view.findViewById(R.id.effectName);
            effectRootView = view.findViewById(R.id.effectsRootView);
        }
    }
}

File: ImageFiltersKotlin/app/src/main/java/org/geeksforgeeks/demo/Adapter.kt
package org.geeksforgeeks.demo

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class Adapter(
    private val items: MutableList<Filters>,
    private val listener: OnFilterClickListener
) :
    RecyclerView.Adapter<Adapter.ViewHolder>() {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val hashTagItemLayoutView = LayoutInflater.from(parent.context)
            .inflate(
                R.layout.item_rv, parent, false
            )
        return ViewHolder(hashTagItemLayoutView)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        val thumbnail = items[position]
        holder.effectTitle.text = thumbnail.title
        holder.effectRootView.setOnClickListener {
            listener.onFilterClicked(thumbnail)
        }
    }

    override fun getItemCount(): Int = items.size

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val effectTitle: TextView = view.findViewById(R.id.effectName)
        val effectRootView: LinearLayout = view.findViewById(R.id.effectsRootView)
    }
}